src/h3lib/lib/baseCells.c

```diff
@@ -31,14 +31,14 @@ typedef struct {
 /** @brief Neighboring base cell ID in each IJK direction.
  *
  * For each base cell, for each direction, the neighboring base
- * cell ID is given. -1 indicates there is no neighbor in that direction.
+ * cell ID is given. 127 indicates there is no neighbor in that direction.
```

Comment:
Just wondering, why is this changed from -1 to 127? Also note that this won't match the generateBaseCellNeighbors program.

Reply:
When you bitmask in -1 for the base cell, it doesn't truncate it to the 8 bits we gave it, and it blows away the hex mode and reserved bits in the upper space of the 64-bit integer.

We could make the `H3_SET_BASE_CELL` macro constrain inputs to the 8-bit range (in which case it would set 255 for a -1 input), but that would slow it down when we could just make it not shove bad data in in the first place.

I chose 127 instead of 255 because I noticed that we're actually only using 7 bits of the 8 bits allocated (I think we allocated 8 bits to keep byte offsets simpler, which makes these bitmasking operations use fewer cycles for most things), and this way if we ever wanted to do anything crazy like mark the upper half of the base cell space for hexagons on mars after Mr. Musk gets us there, then we can. :)

Do you want me to update the `generateBaseCellNeighbors` program to output 127 instead of -1? I was under the impression that output still had to be hand corrected. Did that change?

Reply:
Also, only the neighbor rotation logic used this data, so I determined making changes to it to make the code easier to read and faster was fine.

Reply:
`generateBaseCellNeighbors` does match the contents in baseCells.c exactly, so I think the generator should be updated if we make this change.

One thing I think is nice about -1 is that it's more obviously a sentinel value, while 127 seems like it could be confused with a real base cell ID. If the only use is in `h3NeighborRotations`, would it make sense to simply modify that function to not set an invalid base cell?
```
            int newBaseCell = baseCellNeighbors[oldBaseCell][dir];
            // ...

            if (newBaseCell == -1) {
                // ...
                newBaseCell = // ...;
            }
            H3_SET_BASE_CELL(out, newBaseCell);
```

Reply:
Well, then we introduce runtime latency for no real gain.

I created an `INVALID_BASE_CELL` constant in the codebase, we could use it in this file and fix the generateBaseCellNeighbors to emit that constant and get improved legibility (it is definitely more explicit than -1) with the better perf, too.

Reply:
I'm unclear on there being a performance difference, since the `if (... == INVALID_BASE_CELL` check needs to happen either way (algos.c line 218 in master). Maybe there's something I'm not noticing?

I do agree it makes sense to reuse the INVALID_BASE_CELL constant here and in the generator.

Reply:
I'm not sure about the performance difference, but there would be more instructions needed with this approach for the extra temporary variable and comparison versus reusing the already existing 64-bit register that `out` is stored in and it may or may not require copying a different register out to L1 cache and back in later.

It's not that big of a deal, but I think using the constant variable name is more legible. :)

So I'll update the `baseCells.c` and `generateBaseCellNeighbors` to use it.